### PR TITLE
Improve the description of epoch end

### PR DESCRIPTION
**Pull request by jancionear**
The previous description of when an epoch ends was a bit confusing. It mentioned that two conditions are needed for a block to be the last block of an epoch, but in reality only one of them matters, as one condition is implied by the other. This makes the descripion harder to parse - why are there two conditions, if only one of them matters? It can lead to some confusion.

Let's improve the description to make it clearer and easier to understand. A diagram is also added for clarity.

Refs: [zulip discussion](https://near.zulipchat.com/#narrow/stream/295558-pagoda.2Fcore/topic/When.20does.20an.20Epoch.20end.3F/near/422463251)

### CODE REVIEW

**Comment by jancionear** [comment]
It looks like the existing version of `Epoch.md` didn't pass the linter checks in the CI, so I fixed them.

The `markdown-link-check` fails because some links in unrelated files are throwing `404`, I won't fix them as it's unrelated to this PR.

**Comment by frol** [comment]
@near/wg-protocol Could you, please, review the changes?

**Comment by jancionear** [comment]
> @near/wg-protocol Could you, please, review the changes?

I'd appreciate a review, the PR is over a month old.

File: specs/BlockchainLayer/EpochManager/Epoch.md
Comment (mfornet) [inline specs/BlockchainLayer/EpochManager/Epoch.md:24]:
```suggestion
Note that an epoch only ends when there is a final block above a certain height. If there are no final blocks, the epoch will be stretched until the required final block appears. An epoch can potentially be longer than `epoch_length`.
```

**Review by mfornet — APPROVED** [review]
Looks good to me.

@jancionear, thanks for submitting this change. The diagram is quite helpful.

**Comment by jancionear** [comment]
Github says:
> Code owner review required. Waiting on code owner review from near/nep-moderators

I guess it also needs a review from @near/nep-moderators before merging.

**Comment by bowenwang1996** [comment]
@frol could you approve?

**Review by frol — APPROVED** [review]
Sorry for the delay on my side.

@jancionear Thank you for the contribution and patience! 🙏

**Pull request merged**
